worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -510,16 +510,21 @@ private[celeborn] class Worker(
           // During graceful shutdown, to avoid allocate slots in this worker,
           // add this worker to master's blacklist. When restart, register worker will
           // make master remove this worker from blacklist.
-          rssHARetryClient.send(ReportWorkerUnavailable(List(workerInfo).asJava))
-          val interval = conf.checkSlotsFinishedInterval
           val timeout = conf.checkSlotsFinishedTimeoutMs
-          var waitTimes = 0
-
-          def waitTime: Long = waitTimes * interval
+          val startTime = System.currentTimeMillis()
+          def waitTime: Long = System.currentTimeMillis() - startTime
+          while (waitTime < timeout) {
+            try {
+              rssHARetryClient.send(ReportWorkerUnavailable(List(workerInfo).asJava))
```

Comment:
Seems rssHARetryClient.send will not throw Exception

Reply:
..., It's strange here, I would use askSync instead the send call.